ext/iobuffer/iobuffer.c

```diff
@@ -378,17 +378,22 @@ IO_Buffer_read_from(VALUE self, VALUE io)
 {
     struct buffer  *buf;
     int             ret;
-#if HAVE_RB_IO_T
+#if defined(HAVE_RB_IO_T) || defined(HAVE_RB_IO_DESCRIPTOR)
     rb_io_t        *fptr;
 #else
     OpenFile       *fptr;
 #endif
 
     Data_Get_Struct(self, struct buffer, buf);
-    GetOpenFile(rb_convert_type(io, T_FILE, "IO", "to_io"), fptr);
+    io = rb_convert_type(io, T_FILE, "IO", "to_io");
+    GetOpenFile(io, fptr);
     rb_io_set_nonblock(fptr);
 
+#ifdef HAVE_RB_IO_DESCRIPTOR
+    ret = rb_io_descriptor(io);
```

Comment:
This looks wrong to me.

Reply:
Oops, thanks for catching it. Fixed in https://github.com/socketry/cool.io/pull/79/commits/2112022fe0d8f318aeb7412912ecc63a6ba50bbf.